docs/auth/add-auth-provider.md

```diff
@@ -122,191 +121,128 @@ due to its comprehensive set of supported authentication
-        clientSecret: options.clientSecret,
-        callbackURL: options.callbackUrl,
-        passReqToCallback: false,
-        response_type: 'code',
-        /// ... etc
-      }
-      verifyFunction, // See the "Verify Callback" section
-    );
-  }
-
-  async start() {}
-  async handler() {}
-}
-```
-
-### Adding an non-OAuth based provider
-
-An non-`OAuth` based provider could implement
-`AuthProviderRouteHandlers` instead.
-
-```ts
-type ProviderAOptions = {
-  // ...
-};
-
-export class ProviderAAuthProvider implements AuthProviderRouteHandlers {
-  private readonly _strategy: ProviderAStrategy;
-
-  constructor(options: ProviderAOptions) {
-    this._strategy = new ProviderAStrategy(
-      {
-        // ...
+We're then creating a new module that can extend the Auth backend using the `authProvidersExtensionPoint`.
+
+```ts title="plugins/auth-backend-providername-provider/src/module.ts"
+import { createBackendModule } from '@backstage/backend-plugin-api';
+import {
+  authProvidersExtensionPoint,
+  commonSignInResolvers,
+  createOAuthProviderFactory,
+} from '@backstage/plugin-auth-node';
+import { providerAuthenticator } from './authenticator';
+
+/** @public */
+export const authModuleOktaProvider = createBackendModule({
+  pluginId: 'auth',
+  moduleId: 'providername',
+  register(reg) {
+    reg.registerInit({
+      deps: {
+        providers: authProvidersExtensionPoint,
       },
-      verifyFunction, // See the "Verify Callback" section
-    );
-  }
-
-  async start() {}
-  async frameHandler() {}
-  async logout() {}
-  async refresh() {} // If supported
-}
+      async init({ providers }) {
+        providers.registerProvider({
+          providerId: 'providername',
+          factory: createOAuthProviderFactory({
+            authenticator: providerAuthenticator,
+            signInResolverFactories: {
+              ...commonSignInResolvers,
+            },
+          }),
+        });
+      },
+    });
+  },
+});
 ```
 
-#### Integration Wrapper
-
-Each provider exports an object that provides a way to create new instances
-of the provider, along with related utilities like predefined sign-in resolvers.
-
-The object is created using `createAuthProviderIntegration`, with the most
-important part being the `create` method that acts as the factory function
-for our provider.
-
-The factory should return an implementation of `AuthProviderFactory`, which
-passes in a object with utilities for configuration, logging, token issuing,
-etc. The factory should return an implementation of
-`AuthProviderRouteHandlers`.
-
-The factory is what decides the mapping from
-[static configuration](../conf/index.md) to the creation of auth providers. For
-example, OAuth providers use `OAuthEnvironmentHandler` to allow for multiple
-different configurations, one for each environment, which looks like this;
+Now let's implement the actual authenticator for our provider using `Strategy` from a passport package.
```

Comment:
Having a sentence explaining what authenticators are and listing some other examples in the code base would be nice.

Reply:
It appears that some modules export their authenticators; when should they be reused? I'm updating the OIDC provider documentation, and I am wondering if [I should reuse the OIDC authenticator](https://github.com/backstage/backstage/pull/26394#discussion_r1739621589) or implement a new one every time. I think we can answer this question in this page.